torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -1175,6 +1181,38 @@ def _assert_state(self, state: Union[TrainingState_, List[TrainingState_]]) -> N
                 traceback.print_stack()
             raise ValueError(msg)
 
+    @contextmanager
+    def no_sync(self) -> Generator:
```

Comment:
Can we also file an issue to refactor/deduplicate the no_sync in DDP and FSDP? They look pretty similar and we can probably provide some sort of config / lambda to take care of DDP and FSDP implementation specific details.

Reply:
https://github.com/pytorch/pytorch/issues/72589